holidays/countries/mongolia.py

```diff
@@ -0,0 +1,146 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.constants import PUBLIC, WORKDAY
+from holidays.groups import InternationalHolidays, MongolianCalendarHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Mongolia(HolidayBase, InternationalHolidays, MongolianCalendarHolidays):
+    """Mongolia holidays.
+
+    References:
+        * [Law on Public holidays and days of observation](https://web.archive.org/web/20250327062440/https://legalinfo.mn/mn/detail/399)
+        * [Labor Law](https://web.archive.org/web/20250421093230/https://legalinfo.mn/mn/detail?lawId=16230709635751)
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Mongolia>
+        * [Mongolian lunar calendar](https://web.archive.org/web/20230412171012/https://www.math.mcgill.ca/gantumur/cal/index_mn.html)
+        * <https://web.archive.org/web/20250613054645/https://www.timeanddate.com/holidays/mongolia/>
+        * <https://web.archive.org/web/20250429110535/https://investmongolia.gov.mn/mongolia-at-a-glance/>
+    """
+
+    country = "MN"
+    default_language = "mn"
+    start_year = 2004
+    supported_categories = (PUBLIC, WORKDAY)
+    supported_languages = ("en_US", "mn")
+
+    def __init__(self, *args, **kwargs):
+        InternationalHolidays.__init__(self)
+        MongolianCalendarHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("Шинэ жилийн баяр"))
+
+        # Tsagaan Sar.
+        self._add_tsagaan_sar(tr("Цагаан сар"))
+
+        # Tsagaan Sar Holiday.
+        name = tr("Цагаан сарын баяр")
+        self._add_tsagaan_sar_day_2(name)
+        self._add_tsagaan_sar_day_3(name)
```

Comment:
https://legalinfo.mn/mn/detail?lawId=101888&showType=1
```suggestion
        # Tsagaan Sar Holiday.
        name = tr("Цагаан сарын баяр")
        self._add_tsagaan_sar_day_2(name)
        # Expanded on November 28th, 2013.
        if self._year >= 2014:
            self._add_tsagaan_sar_day_3(name)
```

Reply:
The law doesn't have a separate name for the days of the holiday after the first one. 🙄

Reply:
Yes. The extra holiday word is present in timeanddate.
I'll remove it.

Reply:
Hmm, this is what we did for UAE's Eid al-Fitr day:
```python
        # Eid al-Fitr.
        self._add_eid_al_fitr_day(tr("عيد الفطر"))

        # Eid al-Fitr Holiday.
        eid_al_fitr_holiday = tr("عطلة عيد الفطر")
        self._add_eid_al_fitr_day_two(eid_al_fitr_holiday)
        self._add_eid_al_fitr_day_three(eid_al_fitr_holiday)
        if 2019 <= self._year <= 2024:
            # Ramadan 30 not confirmed yet for 2025 onwards.
            self._add_eid_al_fitr_eve(eid_al_fitr_holiday)
```
```python
    def test_eid_al_fitr(self):
        name = "عيد الفطر"
        name_holiday = "عطلة عيد الفطر"
        self.assertHolidayName(
            name,
            "2017-06-25",
            "2018-06-15",
            "2019-06-04",
            "2020-05-24",
            "2021-05-13",
            "2022-05-02",
            "2023-04-21",
            "2024-04-10",
            "2025-03-30",
        )
        self.assertHolidayName(name, self.no_estimated_holidays, range(1972, 2050))
        self.assertHolidayName(name_holiday, self.no_estimated_holidays, range(1972, 2050))
        self.assertNoHolidayName(name_holiday, "2018-06-14", "2025-03-29")
```
I think simply checking for the first Tsagaan Sar date, and checking for edge-case in day 3's implementation should be enough